src/control/server/storage/config.go

```diff
@@ -532,12 +630,92 @@ func MustNewBdevBusRange(rangeStr string) *BdevBusRange {
 	return br
 }
 
+// OptionBits is a type alias representing option flags as a bitset.
+type OptionBits uint16
+
+type optFlagMap map[string]OptionBits
+
+func (ofm optFlagMap) keys() []string {
+	keys := common.NewStringSet()
+	for k := range ofm {
+		keys.Add(k)
+	}
+
+	return keys.ToSlice()
+}
+
+// toStrings returns a slice of option names that have been set.
+func (obs OptionBits) toStrings(optStr2Flag optFlagMap) []string {
+	opts := common.NewStringSet()
+	for str, flag := range optStr2Flag {
+		if obs&flag == flag {
+			opts.Add(str)
+		}
+	}
+
+	return opts.ToSlice()
+}
+
+// fromStrings generates bitset referenced by the function receiver from the option names provided.
+func (obs *OptionBits) fromStrings(optStr2Flag optFlagMap, opts ...string) error {
+	if obs == nil {
+		return errors.New("fromStrings() called on nil OptionBits")
+	}
+
+	for _, opt := range opts {
+		flag, exists := optStr2Flag[opt]
+		if !exists {
+			return FaultBdevConfigOptFlagUnknown(opt, optStr2Flag.keys()...)
+		}
+		*obs |= flag
+	}
+
+	return nil
+}
+
+// IsEmpty returns true if no options have been set.
+func (obs *OptionBits) IsEmpty() bool {
+	return obs == nil || *obs == 0
+}
+
+var roleOptFlags = optFlagMap{
+	bdevRoleDataName:  BdevRoleData,
+	bdevRoleIndexName: BdevRoleIndex,
+	bdevRoleWALName:   BdevRoleWAL,
+}
+
+// BdevDeviceRoles is a bitset representing SSD role assignments (enabling Metadata-on-SSD).
+type BdevDeviceRoles struct {
+	OptionBits
+}
```

Comment:
I like this, nice work. We should be doing more of this kind of config parsing in the receiver types. My only concern is that I'm not sure the `OptionBits` type needs to be exported.

Reply:
left as is for the moment but I've made a note to revisit